pkg/controllers/core/deployer_controller.go

```diff
@@ -103,11 +102,33 @@ func (r *DeployerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	return result, err
 }
 
+func (r *DeployerReconciler) getAndTrackConfigMap(ctx context.Context, log logr.Logger, deployer *corev1alpha1.Deployer) (*corev1.ConfigMap, error) {
+	cm := &corev1.ConfigMap{}
+	cmKey := types.NamespacedName{Namespace: systemNamespace, Name: settingsConfigMapName}
+
+	// track config map
+	r.Tracker.Track(
+		tracker.NewKey(schema.GroupVersionKind{Version: "v1", Kind: "ConfigMap"}, cmKey),
+		types.NamespacedName{Namespace: deployer.Namespace, Name: deployer.Name},
+	)
+	if err := r.Get(ctx, cmKey, cm); err != nil {
+		log.Error(err, "unable to lookup images configMap")
+		return nil, err
+	}
+
+	return cm, nil
+}
+
 func (r *DeployerReconciler) reconcile(ctx context.Context, log logr.Logger, deployer *corev1alpha1.Deployer) (ctrl.Result, error) {
 	if deployer.GetDeletionTimestamp() != nil {
 		return ctrl.Result{}, nil
 	}
 
+	configMap, err := r.getAndTrackConfigMap(ctx, log, deployer)
```

Comment:
`configMap` is still too generic. What's the semantics role of the config map? The updated name should also be updated within the reconcileIngress and constructIngressForDeployer methods